.github/workflows/unit_test_pipeline.yml

```diff
@@ -0,0 +1,66 @@
+name: TM1Py Integration Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      environments:
+        description: 'JSON array of environments to test (e.g., ["tm1-11-onprem", "tm1-11-cloud"])'
+        required: true
+        default: '["tm1-11-onprem","tm1-11-cloud","tm1-12-mcsp"]'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        environment: ${{ fromJson(github.event.inputs.environments) }}
+    environment: ${{ matrix.environment }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: |
+          pip install -e .[unit_test]
+
+      - name: Retrieve TM1 Connection Details
+        id: get-connection
+        run: echo "Retrieving TM1 connection details"
+        env:
+          TM1_CONNECTION: ${{ vars.TM1_CONNECTION }}
+          TM1_CONNECTION_SECRET: ${{ secrets.TM1_CONNECTION_SECRET }}
+
+      - name: Generate config.ini
+        run: |
+          python Tests/resources/generate_config.py
+        env:
+          TM1_CONNECTION: ${{ vars.TM1_CONNECTION }}
+          TM1_CONNECTION_SECRET: ${{ secrets.TM1_CONNECTION_SECRET }}
+
+      - name: Run tests
+        run: pytest Tests/
```

Comment:
It is actually working like that right now, you can define that in the second parameter when you run the workflow:
<img width="381" alt="image" src="https://github.com/user-attachments/assets/000f444e-73af-407b-9b4a-988aadad2303">
In my case I have 3 environments defined:
<img width="953" alt="image" src="https://github.com/user-attachments/assets/9ea7de4f-1ff6-410c-872e-a342489b3703">

- tm1-11-onprem: a v11 instance we have running in an EC2 instance
- tm1-11-cloud: a v11 instance hosted in IBM PA Cloud
- tm1-12-mcsp: a v12 instance running as SaaS in AWS. 

If I put those three, GH Actions will create 3 different and independent runners on the backend that run **in parallel**. Each of them will have its own config.ini file showing the TM1_CONNECTION and TM1_CONNECTION_SECRET from its own environment. 

Below is a chart explaining the flow:
```mermaid
graph TD
    A[Workflow Dispatch] --> B[Input: Environments]
    B --> C[Parse Environments JSON]
    C --> D{Matrix Strategy}

    %% Parallel jobs for each environment
    D -->|tm1-11-onprem| E1
    D -->|tm1-11-cloud| E2
    D -->|tm1-12-mcsp| E3

    %% Independent Runner for tm1-11-onprem
    subgraph Runner: tm1-11-onprem
        E1[Job: Test tm1-11-onprem] --> F1[Checkout Code]
        F1 --> G1[Set Up Python]
        G1 --> H1[Install Dependencies]
        H1 --> I1[Retrieve TM1 Connection Details]
        I1 --> J1[Generate config.ini]
        J1 --> K1[Run Tests]
    end

    %% Independent Runner for tm1-11-cloud
    subgraph Runner: tm1-11-cloud
        E2[Job: Test tm1-11-cloud] --> F2[Checkout Code]
        F2 --> G2[Set Up Python]
        G2 --> H2[Install Dependencies]
        H2 --> I2[Retrieve TM1 Connection Details]
        I2 --> J2[Generate config.ini]
        J2 --> K2[Run Tests]
    end

    %% Independent Runner for tm1-12-mcsp
    subgraph Runner: tm1-12-mcsp
        E3[Job: Test tm1-12-mcsp] --> F3[Checkout Code]
        F3 --> G3[Set Up Python]
        G3 --> H3[Install Dependencies]
        H3 --> I3[Retrieve TM1 Connection Details]
        I3 --> J3[Generate config.ini]
        J3 --> K3[Run Tests]
    end
```

Reply:
No I mean that the tests itself run in parallel with 'pytest -n auto .\Tests\ElementService_test.py'

Compare #1105

Reply:
Oh! I see, so you re-worked the ElementServce_test.py to be able to run independently from others?
Is this something we can do (or is already done) for other unit test cases? Or you are suggesting to run this one independently and the others sequentially?
Please feel free to create a PR against the yml and put this change in

Reply:
Sorry for being so unprecise. The clue is the -n option of pytest starts 1..n workers. 
At the moment only a few tests can run with more than one worker. But in future it would be beneficial to specify how many workers should run in the pipeline.

Reply:
Running tests in parallel is beneficial when you're working locally and want quick feedback IMO.

However, I'm concerned that parallel tests might occasionally fail due to potential race conditions or non-deterministic behavior in TM1. In the CI pipeline, I would prefer certainty over performance.

On the desktop, where you can easily re-run a failed test, I would prioritize performance over certainty.

Reply:
With the "Matrix Strategy" we should also get benefits of parallelization but the parallel executions should be isolated from one another.